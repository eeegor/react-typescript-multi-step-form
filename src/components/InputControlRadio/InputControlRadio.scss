@import '../../variables';

.input-radio {
	background-color: initial;
	cursor: pointer;
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 1rem;
	height: 0.9rem;
	width: 0.9rem;
	text-align: center;
	display: inline-flex;
	position: relative;
	margin-right: 0.5rem;
}

.input-control-radio {
	display: inline-flex;
	padding: 0 1.5rem;
	margin-right: 0.75rem;
	margin-top: 0.75rem;
	align-items: center;
	justify-items: center;
	border-radius: 2rem;
	color: $color-grey-dark;
	border: 1px solid rgba($color-border, 0.5);
	background-color: white;
	box-shadow: 0 2px 1rem 0 rgba(black, 0.05);
	transition: border-color 0.25s ease-in-out, box-shadow 0.15s ease-in-out;

	.input-control__label {
		font-size: 1rem;
		font-weight: 400;
		position: relative;
		margin-bottom: 0;
		padding: 0.9rem 0 0.75rem 0;
		cursor: pointer;
	}

	&.input-control--checked {
		border-color: $color-success;
		box-shadow: 0 4px 1rem 0 rgba(black, 0.1);

		.input-control__label {
			font-weight: 700;
		}
	}
}
