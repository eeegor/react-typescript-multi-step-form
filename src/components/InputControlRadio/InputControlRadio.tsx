/**
 * Component InputControlRadio
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { InputControl, Label, Info } from '..';
import './InputControlRadio.scss';

export interface InputControlRadioProps {
	id: string;
	name: string;
	info?: string;
	value?: string;
	label?: string;
	required?: boolean;
	autoFocus?: boolean;
	className?: string;
	defaultChecked?: boolean;
	onChange?: (name: string, event: React.ChangeEvent<HTMLInputElement>) => void;
}

export class InputControlRadio extends React.Component<InputControlRadioProps> {
	render(): JSX.Element {
		const {
			id,
			required,
			className,
			value,
			onChange,
			label,
			name,
			defaultChecked,
			info,
			autoFocus,
		} = this.props;

		return (
			<InputControl id={id} type={'radio'} className={classnames(className)}>
				<input
					className="input-control__input input-radio"
					id={id}
					required={required}
					name={name}
					type="radio"
					onChange={event => onChange && onChange(name, event)}
					value={value || ''}
					defaultChecked={defaultChecked || false}
					autoFocus={defaultChecked || autoFocus || false}
				/>
				<Label className="input-control__label" htmlFor={id || name} label={label} />
				<Info className="input-control__info" label={info} />
			</InputControl>
		);
	}
}
