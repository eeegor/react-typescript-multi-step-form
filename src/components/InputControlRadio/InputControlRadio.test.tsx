import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { InputControlRadio } from './InputControlRadio';
import { mount } from 'enzyme';

describe('InputControlRadio', () => {
	const callback = jest.fn();
	const testData = {
		id: 'name',
		name: 'name',
		label: 'Your Name',
		value: 'Bob',
		onChange: callback,
	};

	it('renders markup correctly', () => {
		const component = <InputControlRadio {...testData} />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('change event works', () => {
		const wrapper = mount(<InputControlRadio {...testData} />);
		wrapper.find('.input-control__input').simulate('change');
		expect(callback).toHaveBeenCalledTimes(1);
	});

	it('should adopt a valid info attribute', () => {
		const info = 'test';
		const wrapper = mount(<InputControlRadio info={info} {...testData} />);
		expect(
			wrapper
				.find('.input-control__info')
				.first()
				.text()
		).toEqual(info);
	});
});
