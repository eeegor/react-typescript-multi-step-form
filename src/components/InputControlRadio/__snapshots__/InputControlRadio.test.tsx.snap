// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControlRadio renders markup correctly 1`] = `
<div
  className="input-control input-control-radio"
  id="input-control-name"
>
  <input
    autoFocus={false}
    className="input-control__input input-radio"
    defaultChecked={false}
    id="name"
    name="name"
    onChange={[Function]}
    type="radio"
    value="Bob"
  />
  <label
    className="label input-control__label"
    htmlFor="name"
  >
    Your Name
  </label>
</div>
`;
